boards/arm/bbc_microbit/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2021 Peter Niebert <peter.niebert@univ-amu.fr>
+# SPDX-License-Identifier: Apache-2.0
+
+# add access to board.h
+zephyr_include_directories(.)
```

Comment:
@tejlmand is there currently a similar case elsewhere, and do you think this is the right place to do this or should this be done in core CMake for all boards?

Reply:
not all boards have a `board.h` or header files at all, so i'm not sure we should do this as a general principle.

Also a board could locate its headers in a subdir:
https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/arm/arduino_nano_33_ble/inc

But looking at the `board.h` itself, then I wonder why should pins are not defined in devicetree ?
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/bbc_microbit/board.h
```
#define EXT_P0_GPIO_PIN     3   /* P0, Analog in */
#define EXT_P1_GPIO_PIN     2   /* P1, Analog in */
#define EXT_P2_GPIO_PIN     1   /* P2, Analog in */
.....     
```

Reply:
OK, I agree to all of that.

However, the point of my PR is to change the inclusion rule in CMakefile.txt from the samples to the board directory. The organisation within the board directory is then a matter of the CMakefile.txt and transparent to the rest. I agree it would be better to have somethink like an include directory and maybe a src directory within the board directory.

I also think you are right about your criticism of the file itself. But changing it requires changing all the microbit samples at the same time. I think it would be better to do that in a separate commit?

Now, what I suggest to change right now is to create an include subdirectory and move board.h there. What do you think?

Reply:
On second thought, you are right about dts rather than include file.

So the strategy would be to proceed as follows:

(a) add gpio pin definitions to dts.
(b) change samples to use these rather than board.h definitions
(c) add a depreciation warning to board.h for backward compatibility

Would that be OK?

Reply:
That sounds like a good direction :+1: